storage/innobase/handler/ha_innodb.cc

```diff
@@ -2408,61 +2408,12 @@ innobase_mysql_tmpfile(
 		return(OS_FILE_CLOSED);
 	);
 
-	if (path == NULL) {
-		fd = mysql_tmpfile("ib");
-	} else {
-		fd = mysql_tmpfile_path(path, "ib");
-	}
+	fd = mysql_tmpfile_path(path ? path : mysql_tmpdir, "ib");
 
 	if (fd < 0)
 		return OS_FILE_CLOSED;
```

Comment:
We could remove the local variable `fd` and simply do:
```c
return mysql_tmpfile_path(path ? path : mysql_tmpdir, "ib");
```

Reply:
I didn't get to the point of validating INVALID_HANDLE_VALUE (from OS_FILE_CLOSED) was -1 always.

Reply:
Also separate `mysql_tmpfile_path()` doesn't seem to make any sense.